### ISSUE TITLE
When using bf16_optimizer, there is a doubt about the BF16_Optimizer initially with self.optimizer.step().

### ISSUE THREAD

**Issue by zte-tcb** [microsoft/DeepSpeed#5151]
When I use bf16_optimizer, it will perform self.optimizer.step() at the beginning. I use the Adamw optimizer, that is, a adamw.step(), where it will attenuate params=params*(1-  lr*weight_decay). This leads to the initial weight used in forward at first step of training, And the weight after attenuation used in weight update. That is, the weight of training and weight update is different.

Question 1: Is this incorrect? A weight attenuation performed at the initial of BF16_optimizer;
Question 2: Assuming the question 1 is correct, And I use warmup LR in lr_scheduler. The LR used in this weight attenuation is use Base LR, or the first LR when warmup, which is correct? I know the base LR used by DeepSpeed now, this leads to weight decay greater than the expected.

**Comment by zte-tcb** [comment]
i find this fixed in https://github.com/microsoft/DeepSpeed/pull/5104
